AlexEltonPym/CCC
# Switch to create-react-app

Just sent this via email, but - 

I’ve swapped things over to create-react-app, and made lots of progress on logging in.  I think I may actually have figured the whole shebang out.  I’m going to switch back to my other work for a bit, but I’ll commit the new code back to the repo (under a new branch).

To test:
- run `npm start`
- go to `localtest.me:3000` - that will bring up a login prompt, which will ask for an email address.  Popping one in and hitting go will send a login email to you, with a link
- click the link in your email - that will bring you to `localtest.me:3000/callback`, and you’ll see some console information from the login process.  We’re logged in!


A couple tips:
- at this point, if you need to log out, you have to go into the browser console, go to the Application tab, select “clear storage” on the left, and click the “Clear site data” button that will appear on the right.  This clears out localstorage (and cookies, and a bunch of other crap) which is where the auth0 login credential is stored
- No need to revoke access to test this each time - that was a red herring, I think.  You just need to provide access authorization once per user.  
- it seems like the error with “State” being mismatched was due to us sending the SSO link from the auth0 dashboard.  Confusing, for sure, but sending it from the same URL that we login from seems to have fixed that problem

# 🚨One important thing 🚨 
This branch is currently using an `auth0` app that I set up for myself - to switch to your CCC app, change the settings in `auth0-variables.js`
